### PR TITLE
feat!: Update to SDK V4 & fix infinite loop on transactions

### PR DESCRIPTION
**Pull request by jeromewir**
This PR mainly updates the SDK from V3 to V4

I had to update the tests with some handcrafted data as some JSON structs were empty and returning an error in the tests.
Also omitted the checks for `additional_properties` in `testing.go` as faker cannot populate it (`map[string]interface{}`)

While testing against the sandbox, I also noticed an infinite loop on the `transactions`.
This is due to the cursor still being set but `hasMore` being updated when there is no more data to fetch.

### CODE REVIEW

**Comment by yevgenypats** [comment]
Hey @jeromewir thanks for the PR here! It looks good. 

Few things to be aware of - this repo slipped through the cracks and should've been in archive mode as we don't maintain free plugins anymore (our business is selling plugins while the SDK are open source to enable users to write their own custom sources for private APIs or sources that we don't support yet but users need). 

Im happy to merge this one but we will be archiving it soon and we have it on our roadmap for our official/paid plugins in the near future. You can either fork and upload it as community plugin to `hub.cloudquery.io` or if you would like us to take ownership and wait for the official plugin, happy to reimburse/pay for the work on this PR as it was our bad that we forgot to archive it (feel free to ping me at yp@cloudquery.io).

Let me know your preference, Thanks!

**Comment by jeromewir** [comment]
Hey @yevgenypats thanks for the quick reply!

No worry, it's all good!
Please take ownership, I'll definitely be a paid user once this plugin comes on the hub!
You can close this or merge this as you please.

Thank you for making Cloudquery, love the tool!

**Comment by yevgenypats** [comment]
> Hey @yevgenypats thanks for the quick reply!
> 
> No worry, it's all good! Please take ownership, I'll definitely be a paid user once this plugin comes on the hub! You can close this or merge this as you please.
> 
> Thank you for making Cloudquery, love the tool!

Thanks @jeromewir  !! We will keep you posted here once we have this up and running hopefully pretty shortly.

**Review by murarustefaan — APPROVED** [review]

**Review by erezrokah — APPROVED** [review]
Thanks @jeromewir we'll go ahead and merge this as it looks great. We'll release the next version as a major bump our how Hub

File: client/client.go [hunk 1]
```diff
@@ -19,6 +17,7 @@ type Client struct {
 	ClientId    string
 	Secret      string
 	AccessToken string
+	Spec        Spec
 }
 
 func (c *Client) ID() string {
```
Comment (candiduslynx) [inline client/client.go]:
```suggestion
func (*Client) ID() string {
```

File: client/testing.go [hunk 2]
```diff
@@ -28,49 +29,68 @@ func TestServer(t *testing.T, data any) *httptest.Server {
...
 
+func remove(s schema.ColumnList, i string) schema.ColumnList {
+	for j, c := range s {
+		if c.Name == i {
+			return append(s[:j], s[j+1:]...)
+		}
+	}
+
+	return s
+
```
Comment (candiduslynx) [inline client/testing.go]:
```suggestion
```

File: plugin/plugin.go [hunk 3]
```diff
@@ -1,31 +1,23 @@
...
-	"github.com/cloudquery/plugin-sdk/v3/plugins/source"
-	"github.com/cloudquery/plugin-sdk/v3/schema"
+	"github.com/cloudquery/plugin-sdk/v4/plugin"
 )
 
 var (
 	Version = "development"
+	Kind    = "source"
+	Name    = "example"
+	Team    = "example-team"
```
Comment (candiduslynx) [inline plugin/plugin.go]:
```suggestion
	Name    = "plaid"
	Team    = "cloudquery"
```

**Pull request merged**